contracts/0.4.24/Lido.sol

```diff
@@ -651,6 +673,156 @@ contract Lido is ILido, StETH, AragonApp {
+
+        // Save the current beacon balance and validators to
+        // calculate rewards on the next push
+
+        BEACON_BALANCE_POSITION.setStorageUint256(_beaconBalance);
+
+        if (_beaconValidators > beaconValidators) {
+            BEACON_VALIDATORS_POSITION.setStorageUint256(_beaconValidators);
+        }
+
+        return _beaconValidators.sub(beaconValidators);
+    }
+
+    /**
+     * @dev move funds between ELRewardsVault, deposit and withdrawal buffers. Updates buffered counters respectively
+     */
+    function _processFundsMoving(
+        uint256[] _requestIdToFinalizeUpTo,
+        uint256[] _finalizationPooledEtherAmount,
+        uint256[] _finalizationSharesAmount,
+        uint256 _wcBufferedEther
+    ) internal returns (uint256) {
+        uint256 executionLayerRewards = 0;
+        address elRewardsVaultAddress = getELRewardsVault();
+        // If LidoExecutionLayerRewardsVault address is not set just do as if there were no execution layer rewards at all
+        // Otherwise withdraw all rewards and put them to the buffer
+        if (elRewardsVaultAddress != address(0)) {
+            executionLayerRewards = ILidoExecutionLayerRewardsVault(elRewardsVaultAddress).withdrawRewards(
+                (_getTotalPooledEther() * EL_REWARDS_WITHDRAWAL_LIMIT_POSITION.getStorageUint256()) / TOTAL_BASIS_POINTS
+            );
+        }
+
+        // Moving funds between withdrawal buffer and deposit buffer
+        // depending on withdrawal queue status
+        int256 movedToWithdrawalBuffer = _processWithdrawals(
+            _requestIdToFinalizeUpTo,
+            _finalizationPooledEtherAmount,
+            _finalizationSharesAmount,
+            _wcBufferedEther);
+
+        // Update deposit buffer state
+        if (executionLayerRewards != 0 || movedToWithdrawalBuffer != 0) {
+            if (movedToWithdrawalBuffer > 0) {
+                BUFFERED_ETHER_POSITION.setStorageUint256(
+                    _getBufferedEther().add(executionLayerRewards).add(uint256(movedToWithdrawalBuffer))
+                );
+            } else {
+                BUFFERED_ETHER_POSITION.setStorageUint256(
+                    _getBufferedEther().add(executionLayerRewards).sub(uint256(-movedToWithdrawalBuffer))
+                );
+            }
+        }
+
+        return executionLayerRewards;
+    }
+
+    function _processRewards(
+        uint256 _beaconBalanceNew,
+        uint256 _executionLayerRewards,
+        uint256 _wcBufferedEther,
+        uint256 _beaconBalanceOld,
+        uint256 _appearedValidators
+    ) internal {
+        // Post-withdrawal rewards
+        // rewards = (beacon balance new - beacon balance old) - (appeared validators x 32 ETH)
+        // + withdrawn from execution layer rewards vault + withdrawn from withdrawal credentials vault
+
+        uint256 rewardsBase = (_appearedValidators.mul(DEPOSIT_SIZE)).add(_beaconBalanceOld);
```

Comment:
we are passing two params to calculate the result here, maybe pass `_rewardsBase` to function?

Reply:
We are trying to save some stack in the upper method. And all the `.mul()` and `.add()` invocations take precious stack space here. 
Also, seems right to do calculations where it used.